### ISSUE TITLE
GenerateSerializerSourceCodes doesn't recursive when type is array

### ISSUE THREAD

**Issue by neuecc** [msgpack/msgpack-cli#120]
`SerializerCodeGenerationConfiguration.IsRecursive` is nice feature, thanks.
But if top type is array, doesn't recursive.

```csharp
public class MyClass
{
    // can't generate MyClass3Serializer.
    // If MyClass2 MyProperty, can generate MyClass3Serializer.
    public MyClass2[] MyProperty { get; set; }
}

public class MyClass2
{
    public MyClass3 MyProperty { get; set; }
}

public class MyClass3
{
    public int MyProperty { get; set; }
}
```

repro full code.

```csharp
var settings = new SerializerCodeGenerationConfiguration
{
    OutputDirectory = Path.GetTempPath(),
    SerializationMethod = SerializationMethod.Array,
    Namespace = "Hoge" + ".GeneratedSerializers",
    IsRecursive = true,
    PreferReflectionBasedSerializer = false
};

var result = SerializerGenerator.GenerateSerializerSourceCodes(settings, typeof(MyClass));
foreach (var item in result)
{
    // expected : MyClass, MyClass2, MyClass3
    // actual   : MyClass, MyClass2
    Console.WriteLine(item.TargetType);
}
```

**Comment by yfakariya** [comment]
Sorry, I'll fix it soon.

**Comment by yfakariya** [comment]
It was solved in 0.6.3.
